platform/core/src/runs/TransactingMutationListener.js

```diff
@@ -52,7 +53,8 @@ define([], function () {
                     transactionService.startTransaction();
                 }
 
-                transactionService.addToTransaction(
+                transactionManager.addToTransaction(
```

Comment:
`persistence` is a `TransactionalPersistenceCapability` and it already contains this logic. In fact I suspect everything inside of this `hasChanged` if block could be replaced with one call to `persistence.persist()`? I may be missing some subtlety of behavior here though, our persistence model has gotten way too complicated over the years...

Reply:
This single method call can be replaced, but not the entire hasChanged block. That logic ensures that there is always a transaction open before calling persist. `TransactionalPersistenceCapability` will add to transaction if one exists, otherwise it will persist immediately. 

There are so many layers to old persistence and I can’t wait to refactor it all.

Reply:
Right, but if a transaction is not active it will create one, persist the change, then immediately commit the transaction. I'm questioning why it's necessary to temporarily create a transaction for a single atomic persistence operation? Wouldn't the result of calling `persistenceCapability.commit()` be the same, only without going via a temporary transaction?

Reply:
There's probably a good reason, I just don't remember what it is.

Reply:
@akhenry and that's precisely why I don't want to experiment with it.  I did, however, switch the logic to call persist directly instead of duplicating the addToTransaction logic from persist.

Reply:
Sounds good.